ItemBlockTable tooltip changes

Tiax opened the pull request:
- Show actual item count on ItemBlockTable, when it holds an inventory in NBT.
- Show if PatternChest actually holds patterns or casts, in tooltip.

bonii-xx commented on src/main/java/slimeknights/tconstruct/tools/common/item/ItemBlockTable.java:
```diff
@@ -34,7 +39,40 @@ public void addInformation(@Nonnull ItemStack stack, @Nonnull EntityPlayer playe
+        // determine if it holds casts or patterns
+        boolean holdsCasts = false;
+
+        for (int i = 0; i < items.tagCount(); ++i) {
```
The chest can only hold either casts or patterns, no need for a loop, the first item is enough.

bonii-xx commented on src/main/java/slimeknights/tconstruct/tools/common/item/ItemBlockTable.java:
```diff
@@ -34,7 +39,40 @@ public void addInformation(@Nonnull ItemStack stack, @Nonnull EntityPlayer playe
+        }
+
+        if (holdsCasts) {
+          tooltip.addAll(LocUtils.getTooltips(Util.translateFormatted("tooltip.patternchest.holds_casts", items.tagCount())));
```
Extract format string in variable, set string variable instead of holdsCasts

Tiax commented:
Reworked based on feedback. Still not sure I want to trust the NBT contents blindly - the loop would have ideally only had a single iteration anyway, unless the data is not as expected.

bonii-xx commented:
Well I didn't say that you should trust it blindly, just that we don't need a loop and to make it more readable.

Tiax commented on src/main/java/slimeknights/tconstruct/tools/common/item/ItemBlockTable.java:
```diff
@@ -34,7 +39,27 @@ public void addInformation(@Nonnull ItemStack stack, @Nonnull EntityPlayer playe
+      if (BlockToolTable.TableTypes.fromMeta(stack.getItemDamage()) == BlockToolTable.TableTypes.PatternChest) {
+        // determine if it holds casts or patterns
+        // assume the contents are valid, thus checking any stack in the list suffices
+        Item item = ItemStack.loadItemStackFromNBT(items.getCompoundTagAt(0)).getItem();
```
in which case we'd need the loop again...

Say: slot 0 -> removed item, slot 1 -> valid cast/pattern

Tiax commented:
- Added loop again (still with a single iteration, in a perfect world) to find valid ItemStack to base the holds_casts/holds_patterns on
- rebased the changes into a single commit
